packages/core/src/connectors/index.test.ts

```diff
@@ -0,0 +1,150 @@
+  const connector = connectorMap.get(id);
+
+  if (!connector) {
+    throw new NotFoundError();
+  }
+
+  return connector;
+});
+const hasConnector = jest.fn(async () => true);
+const insertConnector = jest.fn(async (connector: Connector) => connector);
+
+jest.mock('@/queries/connector', () => ({
+  ...jest.requireActual('@/queries/connector'),
+  findConnectorsByIds: async () => connectors,
+  findConnectorById: async (id: string) => findConnectorById(id),
+  hasConnector: async () => hasConnector(),
+  insertConnector: async (connector: Connector) => insertConnector(connector),
+}));
+
+describe('getConnectorInstances', () => {
+  test('should return the connectors existing in DB', async () => {
+    const connectorInstances = await getConnectorInstances();
+    expect(connectorInstances).toHaveLength(connectorInstances.length);
+    expect(connectorInstances[0]).toHaveProperty('connector', aliyunDmConnector);
+    expect(connectorInstances[1]).toHaveProperty('connector', aliyunSmsConnector);
+    expect(connectorInstances[2]).toHaveProperty('connector', facebookConnector);
+    expect(connectorInstances[3]).toHaveProperty('connector', githubConnector);
+    expect(connectorInstances[4]).toHaveProperty('connector', googleConnector);
+  });
+});
+
+describe('getConnectorInstanceById', () => {
+  test('should return the connector existing in DB', async () => {
+    const connectorInstance = await getConnectorInstanceById('aliyun-dm');
+    expect(connectorInstance).toHaveProperty('connector', aliyunDmConnector);
+  });
+
+  test('should throw on invalid id', async () => {
+    const id = 'invalid_id';
+    await expect(getConnectorInstanceById(id)).rejects.toMatchError(
+      new RequestError({ code: 'entity.not_found', id, status: 404 })
+    );
+  });
+});
+
+describe('getSocialConnectorInstanceById', () => {
+  test('should return the connector existing in DB', async () => {
+    const socialConnectorInstance = await getSocialConnectorInstanceById('google');
+    expect(socialConnectorInstance).toHaveProperty('connector', googleConnector);
+  });
+
+  test('should throw on invalid id', async () => {
+    const id = 'invalid_id';
+    await expect(getSocialConnectorInstanceById(id)).rejects.toMatchError(
+      new RequestError({ code: 'entity.not_found', id, status: 404 })
+    );
+  });
+});
+
+describe('getConnectorInstanceByType', () => {
+  test('should return the enabled connector existing in DB', async () => {
+    const dmEnabledConnectorInstance = await getConnectorInstanceByType(ConnectorType.Email);
+    expect(dmEnabledConnectorInstance).toHaveProperty('connector', aliyunDmConnector);
+  });
+
+  test('should throw when there is no enabled connector existing in DB', async () => {
+    const type = ConnectorType.SMS;
+    await expect(getConnectorInstanceByType(type)).rejects.toMatchError(
+      new RequestError('connector.not_found', { type })
+    );
+  });
+});
+
+describe('initConnectors', () => {
+  beforeEach(() => {
```

Comment:
maybe `afterEach`.
Seems more like a clean-up task.

Reply:
No, should be clear before each.
Use mockClear to reset the called times.

Reply:
Add an issue to track this: LOG-1925